### PR TITLE
Refactor create-metric-types.js

### PR DESCRIPTION
**Pull request by santiagopoli**
I've made this refactor to simplify the process of adding new metrics in the future. Based in this change, I'm going to issue a new MR adding the **http_requests_total** metric, which is missing from this library.

### CODE REVIEW

File: packages/metrics/modules/create-metric-types/create-metric-types.js [hunk 1]
```diff
@@ -18,126 +44,85 @@ const defaultMetricTypesOptions = {
...
-          ],
-        }),
-    };
-  }
-
-  return metrics;
+});
+
+const createMetricTypes = options => {
+  let defaultedOptions = { ...defaultOptions, ...options };
```
Comment (tdeekens) [inline packages/metrics/modules/create-metric-types/create-metric-types.js]: Any reason for this not being a const?

File: packages/metrics/modules/create-metric-types/create-metric-types.js [hunk 2]
```diff
@@ -18,126 +44,85 @@ const defaultMetricTypesOptions = {
...
+
+  var defaultMetrics = getDefaultMetrics(defaultedOptions);
```
Comment (tdeekens) [inline packages/metrics/modules/create-metric-types/create-metric-types.js]: These could also be consts to communicate intend a bit better, right?

File: packages/metrics/modules/create-metric-types/create-metric-types.js [hunk 3]
```diff
@@ -18,126 +44,85 @@ const defaultMetricTypesOptions = {
...
+
+  var metricsInMilliseconds =
+    areMetricsInMillisecondsEnabled(defaultedOptions) &&
+    getMetricsInMilliseconds(defaultedOptions);
+
+  var metricsInSeconds =
+    areMetricsInSecondsEnabled(defaultedOptions) &&
+    getMetricsInSeconds(defaultedOptions);
+
+  return { ...defaultMetrics, ...metricsInMilliseconds, ...metricsInSeconds };
```
Comment (tdeekens) [inline packages/metrics/modules/create-metric-types/create-metric-types.js]: This is nice!

**Comment by tdeekens** [comment]
I am not sure we need a total as summary and histograms in Prometheus expose a _count and _sum already. 

- Ref 1: 
https://prometheus.io/docs/practices/histograms/
- Ref 2: 
https://povilasv.me/prometheus-tracking-request-duration

What would you want to achieve with a total that you can not with count and sum?

**Comment by santiagopoli** [comment]
Most prometheus exporters export a metric named **http_requests_total** which is useful to calculate requests per minute, for example.

In the case of requests per minute, the PromQL query would be `round(sum(rate(http_server_requests_total[1m])) * 60)`

Don't know if we can achieve the same result with the provided metrics, but I might be wrong

File: packages/metrics/modules/create-metric-types/create-metric-types.js (hunk 2 above)
Comment (tdeekens) [reply packages/metrics/modules/create-metric-types/create-metric-types.js]: I would appriciate that. Either here or I‘ll do it postmerge. However you prefer.

**Comment by santiagopoli** [comment]
I've made the changes pointed by @tdeekens :D

**Comment by tdeekens** [comment]
I think you should be able to do a: `increase(http_request_duration_buckets_milliseconds_count[5m]))` which should give you what you want. Depending on your goals (e.g. multiple services) you could also `         
sum by (service) (increase(http_request_duration_buckets_milliseconds_count[5m]))`. You query `sum(rate(http_server_requests_count[1m])) * 60)` should also work. Only that I wouldn't `round`. Hope that makes sense.

**Comment by santiagopoli** [comment]
Let's continue this discussion on https://github.com/tdeekens/promster/issues/13

**Pull request merged**